### DISCUSSION TITLE
location configuration failing for proposal configuration when using vici

### DISCUSSION THREAD

**Discussion by vishal1kumar** [strongswan/strongswan#422] [Q&A] [answered]
Stringswan: 5.8.4

I am trying to configure a S2S location using Vici interface. In this I am configuring IKE and ESP Proposals:

```
loc_1= {
    'local_addrs': [],
    'remote_addrs': [],
    'mobike': 'no',
    'proposals': 'aes128-aes192-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp4096-ecp256-ecp384-ecp521',

    'local' : {
        'id': '',
        'auth': 'psk'
    },
    'children' : {
        'child-sa' : {
            'remote_ts': [],
            'mark_in': 0,
            'mark_out': 0,
             'rekey_time': 60,
            'local_ts': ['0.0.0.0/0[tcp/http]', '0.0.0.0/0[tcp/https]'],
             'esp_proposals': 'aes128-aes192-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp4096-ecp256-ecp384-ecp521',
        }
    },
    'remote' : {
        'id': '',
        'auth': 'psk'
    }
}
```
When I am configuring this location using vici, it it throwing below exception for ike proposal:
```
 self.request("load-conn", loc_1)
  File "/usr/local/lib/python3.6/site-packages/vici-5.8.4-py3.6.egg/vici/session.py", line 81, in request
vici.exception.CommandException: Command failed: b'unknown option: proposals, config discarded'
```

Same configuration is working when I am setting it in swanctl.conf for connections {}. Is it an issue with vici or I am doing anything wrong?

**Comment by tobiasbrunner** [comment, marked as answer]
Both of these take lists of proposals, i.e. wrap the strings in `[]`.

**Reply by vishal1kumar** [reply]
> Both of these take lists of proposals, i.e. wrap the strings in `[]`.

ok putting proposals into list like below is not throwing the exception:
` 'proposals': ['aes128-aes192-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp4096-ecp256-ecp384-ecp521']`

**Another question**: how do we indicate in above code include and discard the default algos. In swanctl.conf file I could see we need to do like this:
```
ikev1-pubkey {
        version = 1
        proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-ecp521,aes192-sha256-modp3072,default
```

**Reply by tobiasbrunner** [reply]
What do you mean?

**Reply by Thermi** [reply]
Just add "default" into the list.
E.g. ['aes128-aes192-aes256-sha1-sha256-sha384-sha512-modp1024-modp1536-modp2048-modp4096-ecp256-ecp384-ecp521', 'default']

**Reply by vishal1kumar** [reply]
I missed to reply this earlier, thanks Thermi, ur suggestion worked for me.
